core/services/mongodb-remote/src/test/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizerUnitTests.kt

```diff
@@ -187,4 +993,128 @@ class DataSynchronizerUnitTests {
         // assert that nothing has changed about our state
         assertTrue(dataSynchronizer.isRunning)
     }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun withNewDataSynchronizer(work: (dataSynchronizer: DataSynchronizer,
+                                               coreRemoteMongoCollectionMock: CoreRemoteMongoCollection<BsonDocument>) -> Unit) {
+        val remoteClient = mock(CoreRemoteMongoClientImpl::class.java)
+
+        val dataSynchronizer = spy(DataSynchronizer(
+            instanceKey,
+            service,
+            localClient,
+            remoteClient,
+            networkMonitor,
+            authMonitor
+        ))
+
+        dataSynchronizer.disableSyncThread()
+
+        val databaseSpy = mock(CoreRemoteMongoDatabaseImpl::class.java)
+        `when`(remoteClient.getDatabase(eq(namespace.databaseName))).thenReturn(databaseSpy)
+        val collectionMock = mock(CoreRemoteMongoCollectionImpl::class.java) as CoreRemoteMongoCollectionImpl<BsonDocument>
+        `when`(databaseSpy.getCollection(eq(namespace.collectionName), eq(BsonDocument::class.java))).thenReturn(collectionMock)
+
+        `when`(collectionMock.namespace).thenReturn(namespace)
+        val remoteFindIterable = mock(CoreRemoteFindIterable::class.java) as CoreRemoteFindIterable<BsonDocument>
+        `when`(collectionMock.find(any())).thenReturn(remoteFindIterable)
+        `when`(remoteFindIterable.into<HashSet<BsonDocument>>(any())).thenReturn(HashSet())
+
+        dataSynchronizer.doSyncPass()
+
+        verifyZeroInteractions(collectionMock)
+
+        work(dataSynchronizer, collectionMock)
+    }
+
+    companion object {
+        private fun newDoc(key: String = "hello", value: BsonValue = BsonString("world")): BsonDocument {
+            return BsonDocument("_id", BsonObjectId()).append(key, value)
+        }
+
+        private fun withoutVersionId(document: BsonDocument?): BsonDocument? {
```

Comment:
[heads up] this should be renamed to `withoutSyncVersion` to be consistent with the change I made in my PR